router/worker.go

```diff
@@ -306,14 +329,11 @@ func (w *worker) transform(routerJobs []types.RouterJobT) []types.DestinationJob
 			w.rt.logger.Debugn("traceParent is empty during router transform", logger.NewIntField("jobId", job.JobMetadata.JobID))
 		}
 	}
-	w.rt.routerTransformInputCountStat.Count(len(routerJobs))
-	destinationJobs := w.rt.transformer.Transform(
-		transformer.ROUTER_TRANSFORM,
-		&types.TransformMessageT{Data: routerJobs, DestType: strings.ToLower(w.rt.destType)},
-	)
-	w.rt.routerTransformOutputCountStat.Count(len(destinationJobs))
-	w.recordStatsForFailedTransforms("routerTransform", destinationJobs)
-	return destinationJobs
+
+	if w.rt.reloadableConfig.oauthV2Enabled.Load() && w.rt.isOAuthDestination {
```

Comment:
no need to check whether `oauthV2Enabled` if it is not an oauth destination
```suggestion
	if w.rt.isOAuthDestination && w.rt.reloadableConfig.oauthV2Enabled.Load() {
```